pkg/proxy/handler.go

```diff
@@ -83,13 +87,36 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	}
 
 	log.Println("Waiting for IPs", id)
-	host := h.Endpoints.GetHost(r.Context(), deploy, "http")
+	host, err := h.Endpoints.AwaitHostAddress(r.Context(), deploy, "http")
+	if err != nil {
+		log.Printf("timeout finding the host address %v", err)
+		w.WriteHeader(http.StatusRequestTimeout)
```

Comment:
Did I understand it correctly that right now there is no timeout configured? Only the test cases do set a timeout?

Reply:
There is no explicit timeout defined. But it can make sense to define this at a higher level for all requests.

Reply:
agreed, let's capture this in a separate issue, I can file it. I think a default timeout of 30 minutes would make sense with an option to set it lower

Reply:
I filed this issue: https://github.com/substratusai/lingo/issues/58